public/lesswrong-editor/src/internal-block-links.js

```diff
@@ -0,0 +1,30 @@
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+
+export default class InternalBlockLinks extends Plugin {
+  init() {
+    const editor = this.editor;
+
+    // Extend the schema for block elements to allow a `data-internal-id` attribute
+    editor.model.schema.extend('$block', {
+      allowAttributes: ['data-internal-id']
+    });
```

Comment:
You may well ask "why not just allow the id attribute?". The main practical reason is that I want to have both a data-internal-id and an id on headings. The id might change if they edit the heading after importing, but links to the data-internal-id will keep working. It also technically fixes the problem of duplicate ids, but this is unlikely to happen anyway